### PR TITLE
LPS-78486 forms

### PR DESCRIPTION
**Pull request by javiergamarra**
I suppose I should split both identifier in 2 modules :S (I don't want to have more modules) and there is some unused code (custom actions) that I left there hoping custom actions is merged soon...

### CODE REVIEW

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceRecordNestedCollectionResource.java [hunk 1]
```diff
@@ -210,30 +211,18 @@ private Long _extractFileEntryId(DDMFormFieldValue ddmFormFieldValue) {
...
-		if (count == null) {
-			return null;
-		}
+		Pagination pagination, Long formInstanceId) throws PortalException {
 
-		return Try.fromFallible(
-			() -> _ddmFormInstanceRecordService.getFormInstanceRecords(
+		List<DDMFormInstanceRecord> ddmFormInstanceRecords =
+			_ddmFormInstanceRecordService.getFormInstanceRecords(
 				formInstanceId, WorkflowConstants.STATUS_ANY,
```
Comment (eperezg) [inline modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceRecordNestedCollectionResource.java]: STATUS_ANY ¿?

File: modules/apps/headless-apio/media-object/media-object-apio-impl/src/main/java/com/liferay/media/object/apio/internal/architect/resource/MediaObjectNestedCollectionResource.java [hunk 2]
```diff
@@ -68,7 +68,7 @@
 		return builder.addGetter(
 			this::_getPageItems
 		).addCreator(
-			this::_addFileEntry,
```
Comment (eperezg) [inline modules/apps/headless-apio/media-object/media-object-apio-impl/src/main/java/com/liferay/media/object/apio/internal/architect/resource/MediaObjectNestedCollectionResource.java]: keep add ¿?

File: modules/apps/headless-apio/structure/structure-apio-api/bnd.bnd [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Bundle-Name: Liferay Structure Apio API
+Bundle-SymbolicName: com.liferay.structure.apio.api
+Bundle-Version: 2.0.0
```
Comment (eperezg) [inline modules/apps/headless-apio/structure/structure-apio-api/bnd.bnd:3]: 2.0.0 ¿?

File: modules/apps/headless-apio/structure/structure-apio-impl/bnd.bnd [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Bundle-Name: Liferay Structure Apio Implementation
+Bundle-SymbolicName: com.liferay.structure.apio.impl
+Bundle-Version: 2.0.0
```
Comment (eperezg) [inline modules/apps/headless-apio/structure/structure-apio-impl/bnd.bnd:3]: 2.0.0 ¿?

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/helper/EvaluateContextHelper.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+			FormContextWrapper::new
+		).orElse(
+			null
+		);
+	}
+
+	private void _setEvaluatorLocale(
+		DDMFormRenderingContext ddmFormRenderingContext, Locale locale) {
+
+		LocaleThreadLocal.setThemeDisplayLocale(locale);
```
Comment (eperezg) [inline modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/helper/EvaluateContextHelper.java:80]: threadLocal ¿?

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java [hunk 6]
```diff
@@ -163,58 +124,75 @@ public String getName() {
...
+			"FormInstanceSettings"
+		).addBoolean(
+			"isPublished", DDMFormInstanceSettings::published
+		).addBoolean(
+			"isRequireAuthentication",
+			DDMFormInstanceSettings::requireAuthentication
+		).addBoolean(
+			"isRequireCaptcha", DDMFormInstanceSettings::requireCaptcha
+		).addNested(
+			"emailNotification", settings -> settings,
```
Comment (eperezg) [inline modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java:145]: settings -> settings ¿?

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceRecordNestedCollectionResource.java [hunk 7]
```diff
@@ -109,98 +117,94 @@ public String getName() {
...
-		else {
-			serviceContext.setWorkflowAction(WorkflowConstants.ACTION_PUBLISH);
-		}
+	private PageItems<DDMFormInstanceRecord> _getPageItems(
+			Pagination pagination, Long formInstanceId)
+		throws PortalException {
+
+		List<DDMFormInstanceRecord> ddmFormInstanceRecords =
+			_ddmFormInstanceRecordService.getFormInstanceRecords(
+				formInstanceId, WorkflowConstants.STATUS_ANY,
```
Comment (eperezg) [inline modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceRecordNestedCollectionResource.java:193]: STATUS_ANY ¿?

**Comment by eperezg** [comment]
Also maybe we should use _addFileEntry instead of _getFileEntry on  https://github.com/eperezg/liferay-portal/pull/28/commits/f9c38f448a103ce4515a1830027e3f00e66c35f0

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceRecordNestedCollectionResource.java (hunk 1 above)
Comment (javiergamarra) [reply modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceRecordNestedCollectionResource.java]: if (status == WorkflowConstants.STATUS_ANY) {
				sql = StringUtil.replace(
					sql, "(DDMFormInstanceRecordVersion.status = ?) AND",
					StringPool.BLANK);
			}

File: modules/apps/headless-apio/media-object/media-object-apio-impl/src/main/java/com/liferay/media/object/apio/internal/architect/resource/MediaObjectNestedCollectionResource.java (hunk 2 above)
Comment (javiergamarra) [reply modules/apps/headless-apio/media-object/media-object-apio-impl/src/main/java/com/liferay/media/object/apio/internal/architect/resource/MediaObjectNestedCollectionResource.java]: 👍

File: modules/apps/headless-apio/structure/structure-apio-api/bnd.bnd (hunk 3 above)
Comment (javiergamarra) [reply modules/apps/headless-apio/structure/structure-apio-api/bnd.bnd:3]: 👍

File: modules/apps/headless-apio/structure/structure-apio-impl/bnd.bnd (hunk 4 above)
Comment (javiergamarra) [reply modules/apps/headless-apio/structure/structure-apio-impl/bnd.bnd:3]: 👍

File: modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java (hunk 6 above)
Comment (javiergamarra) [reply modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java:145]: 👍

**Pull request closed without merging**